instance/gandi.py

```diff
@@ -52,50 +52,51 @@ def api_key(self):
         return settings.GANDI_API_KEY
 
     @property
-    def zone_id(self):
-        """
-        Gandi Zone ID of the domain
-        """
-        return settings.GANDI_ZONE_ID
-
-    @property
     def client_zone(self):
         """
         Client domain zone API endpoint
         """
         return self.client.domain.zone
 
-    def delete_dns_record(self, zone_version_id, record_name):
+    def get_zone_id(self, domain):
+        """
+        Gandi zone ID used by domain
+        """
+        return self.client.domain.info(self.api_key, domain)['zone_id']
```

Comment:
Since the result of this call will never change, it might make sense to cache it.  Since I don't think `functools.lru_cache` will work here, I'd simply store the zone IDs in a dictionary.